azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1776,12 +1776,49 @@ private void processFlow(ExecutionReference reference,
       }
     }
 
+    /* Helper method to fetch  overriding Executor, if a valid user has specifed otherwise return null */
+    private Executor getUserSpecifiedExecutor(ExecutableFlow exflow) {
```

Comment:
an aesthetic choice:  Do you think we should just pass in the ExecutionOptions object? This method doesn't seem to require other things in the exflow.